File: leakcanary-analyzer/src/main/java/com/squareup/leakcanary/AnalyzerProgressListener.java
package com.squareup.leakcanary;

import android.support.annotation.NonNull;

public interface AnalyzerProgressListener {

  @NonNull AnalyzerProgressListener NONE = new AnalyzerProgressListener() {
    @Override public void onProgressUpdate(@NonNull Step step) {
    }
  };

  // These steps should be defined in the order in which they occur.
  enum Step {
    READING_HEAP_DUMP_FILE,
    PARSING_HEAP_DUMP,
    DEDUPLICATING_GC_ROOTS,
    FINDING_LEAKING_REF,
    FINDING_SHORTEST_PATH,
    BUILDING_LEAK_TRACE,
    COMPUTING_DOMINATORS,
    COMPUTING_BITMAP_SIZE,
  }

  void onProgressUpdate(@NonNull Step step);
}

File: leakcanary-analyzer/src/main/java/leakcanary/AnalyzerProgressListener.kt
package leakcanary

/**
 * Used to report progress by the [HeapAnalyzer].
 */
interface AnalyzerProgressListener {

  // These steps are defined in the order in which they occur.
  enum class Step {
    PARSING_HEAP_DUMP,
    FINDING_LEAKING_INSTANCES,
    FINDING_PATHS_TO_LEAKING_INSTANCES,
    FINDING_DOMINATORS,
    COMPUTING_NATIVE_RETAINED_SIZE,
    COMPUTING_RETAINED_SIZE,
    BUILDING_LEAK_TRACES,
  }

  fun onProgressUpdate(step: Step)

  companion object {
    val NONE = object : AnalyzerProgressListener {
      override fun onProgressUpdate(step: Step) {
      }
    }
  }
}